src/util/resolveDefaultProps.tsx

```diff
@@ -0,0 +1,24 @@
+/**
+ * This function mimics the behavior of the `defaultProps` static property in React.
+ * Functional components do not have a defaultProps property, so this function is useful to resolve default props.
+ *
+ * The common recommendation is to use ES6 destructuring with default values in the function signature
```

Comment:
This was more of a problem when there was element cloning imo. I guess there still is quite a bit of that